### PR TITLE
Retire dashboard v1 (js and python)

### PR DESCRIPTION
**Pull request by graceguo-supercat**
this PR depends on #5463.

Remove dashboard v1-only and v1 => v2 transition related logic, and update unit tests.

build passed 😂

@williaster @mistercrunch

### CODE REVIEW

File: superset/assets/src/dashboard/components/Dashboard.jsx [hunk 1]
```diff
@@ -86,9 +86,7 @@ class Dashboard extends React.PureComponent {
 
   componentWillReceiveProps(nextProps) {
     if (!nextProps.dashboardState.editMode) {
-      const version = nextProps.dashboardState.isV2Preview
-        ? 'v2-preview'
-        : 'v2';
+      const version = 'v2';
```
Comment (williaster) [inline superset/assets/src/dashboard/components/Dashboard.jsx]: nit I wouldn't define a variable for this, you can just define it as part of the object below.

File: superset/assets/src/dashboard/components/SaveModal.jsx [hunk 2]
```diff
@@ -104,16 +103,12 @@ class SaveModal extends React.PureComponent {
...
     return (
       <ModalTrigger
         ref={this.setModalRef}
         isMenuItem={this.props.isMenuItem}
         triggerNode={this.props.triggerNode}
-        modalTitle={t(
-          'Save Dashboard%s',
-          isV2Preview ? ' (⚠️ all saved dashboards will be V2)' : '',
-        )}
+        modalTitle={t('Save Dashboard%s')}
```
Comment (williaster) [inline superset/assets/src/dashboard/components/SaveModal.jsx]: extra `%` I think?

File: superset/views/core.py [hunk 3]
```diff
@@ -2140,59 +2101,11 @@ def dashboard(self, dashboard_id):
...
-                if force_v2_edit:
-                    dash_edit_perm = False
-
         # Hack to log the dashboard_id properly, even when getting a slug
         @log_this
         def dashboard(**kwargs):  # noqa
             pass
 
-        # TODO remove extra logging upon v1 deprecation 🎉
-        dashboard(
```
Comment (williaster) [inline superset/views/core.py]: I think you still need to call the `dashboard()` method or there will be no logging. just need to remove the v1/2-switch-specific `kwargs`.

**Review by williaster — COMMENTED** [review]
couple minor changes but lgtm overall! this is exciting! 🎉 

Not sure how you are thinking about merging things but I feel like frontend deprecation shouldn't be merged until _after_ the python migration script is in place. wdyt @john-bodley @mistercrunch ?

**Comment by williaster** [comment]
I wonder also if we should delete the markdown + separator slice types?

File: superset/assets/src/dashboard/components/Dashboard.jsx (hunk 1 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: fixed.

File: superset/assets/src/dashboard/components/SaveModal.jsx (hunk 2 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/components/SaveModal.jsx]: fixed. removed `%s`

**Comment by graceguo-supercat** [comment]
i think if user ever saved a `separator` or `markdown` slice before, they should still be able to access by slice_id or etc. Also for geo related viz, we allow user give custom js code as render function. User define their js code in markdown type viz. 

So I didn't delete those viz type, but i added a logic in `Save to dashboard`. if current viz_type is `separator` or `markdown`, the save to dash option is disabled (so that this slice can't be added to dashboard from explore view).

File: superset/views/core.py (hunk 3 above)
Comment (graceguo-supercat) [reply superset/views/core.py]: fixed.

File: superset/assets/package.json [hunk 4]
```diff
@@ -102,7 +102,6 @@
     "react-dnd-html5-backend": "^2.5.4",
     "react-dom": "^15.6.2",
     "react-gravatar": "^2.6.1",
-    "react-grid-layout": "0.16.6",
```
Comment (williaster) [inline superset/assets/package.json]: 🎉

**Review by williaster — APPROVED** [review]
This LGTM, I would still say the DB migration (in #5463) should be merged before this tho otherwise folks might view an unconverted v1 dashboard which would fail.

**Pull request merged**